### PR TITLE
Added package distribution support for mariner 3.0

### PR DESCRIPTION
**Pull request by dphulkar-msft**
Tested preview package publish:
x86_64: https://packages.microsoft.com/azurelinux/3.0/preview/ms-oss/x86_64/Packages/a/
aarch64: https://packages.microsoft.com/azurelinux/3.0/preview/ms-oss/aarch64/Packages/a/

### CODE REVIEW

**Review by souravgupta-msft — APPROVED** [review]

File: release-pipeline.yml [hunk 1]
```diff
@@ -1928,7 +1928,7 @@ stages:
                     if [[ $archetype == *"Arm"* ]]; then
                       echo "Skipping for ARM type on $distro"
                     else
-                      if [[ $distro == *"Mariner-2.0"* ]]; then
+                      if [[ $distro == *"Mariner-2.0"* || $distro == *"Mariner-3.0"* ]]; then
```
Comment (vibhansa-msft) [inline release-pipeline.yml:1931]: We shall rather check for "Mariner-" instead of checking 2.0 and 3.0

File: release-pipeline.yml [hunk 2]
```diff
@@ -1910,7 +1910,7 @@ stages:
                 while IFS=, read -r distro archetype repoName releaseName; do
 
                   # If the package is preview, publish to mariner preview package
-                  if [[ $distro == *"Mariner-2.0"* ]]; then
+                  if [[ $distro == *"Mariner-2.0"* || $distro == *"Mariner-3.0"* ]]; then
```
Comment (vibhansa-msft) [inline release-pipeline.yml:1913]: Just check "Mariner-"

File: release-pipeline.yml (hunk 1 above)
Comment (dphulkar-msft) [reply release-pipeline.yml]: done

File: release-pipeline.yml (hunk 2 above)
Comment (dphulkar-msft) [reply release-pipeline.yml]: done

**Review by vibhansa-msft — APPROVED** [review]

**Pull request merged**